src/jshint.js

```diff
@@ -2496,7 +2496,7 @@ var JSHINT = (function() {
     if (left) {
       if (left.type === "(identifier)") {
         if (left.value.match(/^[A-Z]([A-Z0-9_$]*[a-z][A-Za-z0-9_$]*)?$/)) {
-          if ("Number String Boolean Date Object Error Symbol".indexOf(left.value) === -1) {
+          if ("Array Number String Boolean Date Object Error Symbol".indexOf(left.value) === -1) {
```

Comment:
I think I do agree with the OP for the referenced bug though, doing this using `indexOf` is a bit iffy, because it means `BigNumber` or `LocalizedString` or `MyFancyError` are effected as well.

Probably a good idea to make a map in vars.js like

```js
var kCallableConstructors = Object.setPrototypeOf({
  Array: true,
  Number: true,
  String: true,
  Boolean: true,
  Date: true,
  Object: true,
  Symbol: true,
  Error: true,
  TypeError: true,
  SyntaxError: true,
  ReferenceError: true,
  // ETC for all the crazy DOM errors and such
}, null);
```